### PR TITLE
ci: add new jobs for deploy storybook

### PR DESCRIPTION
**Pull request by mikqi**
## Hey, I just made a Pull Request for issue https://github.com/spotify/backstage/issues/365!

Add new jobs when new commit on master it will build and deploy to gh-pages.
This job use [action/deploy-to-github-pages](https://github.com/marketplace/actions/deploy-to-github-pages) need actions to add `secrets/ACCESS_TOKEN` at settings.

#### :heavy_check_mark: Checklist

- [ ] All tests are passing `yarn test`
- [ ] Screenshots attached (for UI changes)
- [ ] Relevant documentation updated
- [ ] Prettier run on changed files
- [ ] Tests added for new functionality
- [ ] [CODEOWNERS](./CODEOWNERS) updated (for new stuff)
- [ ] Regression tests added for bug fixes

### CODE REVIEW

File: .github/workflows/frontend.yml [hunk 1]
```diff
@@ -3,6 +3,8 @@ name: Frontend CI
 on:
   pull_request:
     types: [opened, reopened, edited, synchronize]
+  push:
+    branches: master
```
Comment (Rugvip) [inline .github/workflows/frontend.yml]: Can we make it a separate workflow that only triggers on changes to `packages/core`? :>

File: .github/workflows/frontend.yml [hunk 2]
```diff
@@ -40,3 +42,22 @@ jobs:
...
+        node-version: [12.x]
+    steps:
+      - uses: actions/checkout@v2
+      - name: build storybook'
+        run: |
+          yarn build-storybook
+      - name: deploy gh-pages
+        uses: JamesIves/github-pages-deploy-action@3.4.2
+        with:
+          ACCESS_TOKEN: ${{ secrets.ACCESS_TOKEN }}
```
Comment (Rugvip) [inline .github/workflows/frontend.yml]: Please switch this to GITHUB_TOKEN. I've created a gh-pages branch, which should work around the issue with publishing in public repositories.

File: .github/workflows/frontend.yml [hunk 3]
```diff
@@ -3,6 +3,11 @@ name: Frontend CI
 on:
   pull_request:
     types: [opened, reopened, edited, synchronize]
+  push:
+    branches:
+      - master
+    paths:
+      - 'packages/core/src/**'
```
Comment (Rugvip) [inline .github/workflows/frontend.yml]: 👍

File: .github/workflows/frontend.yml [hunk 4]
```diff
@@ -42,3 +47,23 @@ jobs:
           yarn test
         env:
           CI: true
+
+  deploy-storybook:
+    if: github.event_name == 'push' && github.ref == 'refs/heads/master'
```
Comment (Rugvip) [inline .github/workflows/frontend.yml]: So I think we still want to build the storybook in PRs as well, but only publish from master, although we can add PR builds later. Either way, I'd move this entire section into a new workflow file and leave this one as-is. The problem with having it here is that we now run the above job on master builds too, which we don't really want, we have a separate workflow for that.

File: .github/workflows/frontend.yml [hunk 5]
```diff
@@ -42,3 +47,23 @@ jobs:
...
+    steps:
+      - uses: actions/checkout@v2
+      - name: build storybook'
+        run: |
+          yarn install
+          yarn build-storybook
+      - name: deploy gh-pages
+        uses: JamesIves/github-pages-deploy-action@3.4.2
+        with:
+          ACCESS_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (Rugvip) [inline .github/workflows/frontend.yml]: The parameter name should also be switched to GITHUB_TOKEN

File: .github/workflows/frontend.yml [hunk 6]
```diff
@@ -42,3 +47,23 @@ jobs:
...
+          BRANCH: gh-pages
+          FOLDER: packages/storybook/dist
```
Comment (Rugvip) [inline .github/workflows/frontend.yml]: 👍

File: package.json [hunk 7]
```diff
@@ -10,8 +10,8 @@
     "test": "cross-env CI=true lerna run test --since origin/master -- --coverage",
     "create-plugin": "backstage-cli create-plugin",
     "lint": "lerna run lint",
-    "storybook": "yarn workspace @spotify-backstage/core storybook",
-    "build-storybook": "yarn workspace @spotify-backstage/core build-storybook"
+    "storybook": "yarn workspace storybook start",
```
Comment (Rugvip) [inline package.json]: Just submitted #380 to fix this 🙈 , also got rid of the top-level build command though, for reasons mentioned in the PR. Should be alright to just run the build-storybook script inside the storybook package?

**Review by Rugvip — COMMENTED** [review]
Bombarding you with comments, but thank you for this work :grin:❤️ Gonna be awesome to have always up-to-date storybook deployed.

File: package.json (hunk 7 above)
Comment (mikqi) [reply package.json]: it's alright. 👍🏻
so, i'll update the build script in config.

File: .github/workflows/frontend.yml (hunk 4 above)
Comment (mikqi) [reply .github/workflows/frontend.yml]: okey got it @Rugvip 👍🏻

**Comment by stefanalund** [comment]
@mikqi @Rugvip what’s the status of this PR? Really excited about getting the Storybook published on storybook.backstage.io 😍

**Comment by Rugvip** [comment]
Need the jobs to be separated, but we could also merge this and fix in a followup PR

**Review by Rugvip — DISMISSED** [review]

**Comment by stefanalund** [comment]
@mikqi wanna fix the merge conflict, and then we can merge? Thanks

**Comment by mikqi** [comment]
> @mikqi wanna fix the merge conflict, and then we can merge? Thanks

Fixed. 👍🏻

**Review by stefanalund — APPROVED** [review]

**Review by freben — APPROVED** [review]

**Pull request merged**